chips/stm32f4xx/src/clocks/pll.rs

```diff
@@ -304,16 +331,22 @@ impl<'a, PllConstants: clock_constants::PllConstants> Pll<'a, PllConstants> {
         // PLL output frequency = VCO output frequency / PLLP
         // PLL48CLK = VCO output frequency / PLLQ
 
+        // Get source frequency
+        let source_frequency = self.get_pll_source_frequency();
+
+        // Set source (HSI or HSE)
+        self.set_pll_source_clock();
```

Comment:
If I understand the code correctly, the PLL will choose the suitable source depending on the desired output frequency, right? And why is the source frequency first retrieved, then the source of PLL changed?

Reply:
The PLL need to know the source frequency. it can be either HSI or HSE. So first the PLL ask what is the source frequency (for PLLP/N/Q calculations).
The second step is to set the pll source according to the system clock

Reply:
This answer is related to [this comment](https://github.com/tock/tock/pull/3750#discussion_r1406164657): it looks like you tied PLL source clock to the system clock. I don't think this is a good idea. Imagine the following situation:

1. HSI is the system clock to provide a default 16MHz frequency to most peripherals.
2. HSE is the PLL source clock to provide a 48MHz frequency to the random number generator peripheral.

You can see that depending on one's needs, the PLL source clock can differ from system clock source.

What do you think? Did I miss something?

Reply:
I get what you are saying. will try to fix it.

Reply:
I believe it is now fixed